source/Octopus.Tentacle.Client/Decorators/HalibutExceptionTentacleServiceDecorator.cs

```diff
@@ -1,222 +1,58 @@
-        {
-            return new HalibutExceptionScriptServiceV2Decorator(service);
-        }
-
-        public IAsyncClientScriptServiceV2 Decorate(IAsyncClientScriptServiceV2 service)
-        {
-            return new HalibutExceptionAsyncScriptServiceV2Decorator(service);
-        }
-
-        public IClientFileTransferService Decorate(IClientFileTransferService service)
-        {
-            return service;
-        }
-
-        public IAsyncClientFileTransferService Decorate(IAsyncClientFileTransferService service)
-        {
-            return service;
-        }
-
-        public IClientCapabilitiesServiceV2 Decorate(IClientCapabilitiesServiceV2 service)
-        {
-            return new HalibutExceptionCapabilitiesServiceV2Decorator(service);
-        }
-
-        public IAsyncClientCapabilitiesServiceV2 Decorate(IAsyncClientCapabilitiesServiceV2 service)
-        {
-            return new HalibutExceptionAsyncCapabilitiesServiceV2Decorator(service);
-        }
-
-        public static bool IsHalibutOperationCancellationException(Exception e)
-        {
-            return e is HalibutClientException && e.Message.Contains("The operation was canceled");
-        }
-    }
-
-    class HalibutExceptionScriptServiceV2Decorator : IClientScriptServiceV2
-    {
-        private readonly IClientScriptServiceV2 inner;
-
-        public HalibutExceptionScriptServiceV2Decorator(IClientScriptServiceV2 inner)
-        {
-            this.inner = inner;
-        }
-
-        public ScriptStatusResponseV2 StartScript(StartScriptCommandV2 command, HalibutProxyRequestOptions halibutProxyRequestOptions)
-        {
-            try
-            {
-                return inner.StartScript(command, halibutProxyRequestOptions);
-            }
-            catch (Exception e) when (HalibutExceptionTentacleServiceDecorator.IsHalibutOperationCancellationException(e))
-            {
-                throw new OperationCanceledException("The operation was cancelled", e);
-            }
-        }
-
-        public ScriptStatusResponseV2 GetStatus(ScriptStatusRequestV2 request, HalibutProxyRequestOptions halibutProxyRequestOptions)
-        {
-            try
-            {
-                return inner.GetStatus(request, halibutProxyRequestOptions);
-            }
-            catch (Exception e) when (HalibutExceptionTentacleServiceDecorator.IsHalibutOperationCancellationException(e))
-            {
-                throw new OperationCanceledException("The operation was cancelled", e);
-            }
-        }
-
-        public ScriptStatusResponseV2 CancelScript(CancelScriptCommandV2 command, HalibutProxyRequestOptions halibutProxyRequestOptions)
+        protected static Task<TResponse> HandleCancellationException<TResponse>(Func<Task<TResponse>> action)
         {
             try
             {
-                return inner.CancelScript(command, halibutProxyRequestOptions);
+                return action();
```

Comment:
You need to await this or your exceptions won't be caught properly.

Reply:
and that's why my test was failing 😂 
Thanks!